### ISSUE TITLE
Panic when running cargo watch

### ISSUE THREAD

**Issue by rschulman** [passcod/cargo-watch#50]
I installed cargo-watch using `cargo install ...` and when I try to run it, it panics trying to unwrap a Result. On a mac, Mac OS X Yosemite.

 Backtrace:
```
✗ RUST_BACKTRACE=1 cargo watch
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { repr: Os { code: 2, message: "No such file or directory" } }', ../src/libcore/result.rs:799
stack backtrace:
   1:        0x10fb4a1a8 - std::sys::backtrace::tracing::imp::write::h6f1d53a70916b90d
   2:        0x10fb4e6ff - std::panicking::default_hook::{{closure}}::h137e876f7d3b5850
   3:        0x10fb4d705 - std::panicking::default_hook::h0ac3811ec7cee78c
   4:        0x10fb4dd16 - std::panicking::rust_panic_with_hook::hc303199e04562edf
   5:        0x10fb4dbb4 - std::panicking::begin_panic::h6ed03353807cf54d
   6:        0x10fb4dad2 - std::panicking::begin_panic_fmt::hc321cece241bb2f5
   7:        0x10fb4da37 - rust_begin_unwind
   8:        0x10fb73f60 - core::panicking::panic_fmt::h27224b181f9f037f
   9:        0x10fa90320 - core::result::unwrap_failed::hbbe45964de00535c
  10:        0x10fa9db9d - cargo_watch::main::hd1e92082f03362ed
  11:        0x10fb4ecba - __rust_maybe_catch_panic
  12:        0x10fb4d246 - std::rt::lang_start::h538f8960e7644c80
```

**Comment by passcod** [comment]
Can you give a directory listing of where you ran it? `ls -shal` will do

**Comment by rschulman** [comment]
Certainly... here you go:

```
➜  northship git:(websocket_focus) ✗ ls -shal
total 64
 0 drwxr-xr-x   9 schulmanr  staff   306B Dec 28 22:01 .
 0 drwxr-xr-x  55 schulmanr  staff   1.8K Dec 28 21:43 ..
 0 drwxr-xr-x  12 schulmanr  staff   408B Dec 28 22:27 .git
 8 -rw-r--r--   1 schulmanr  staff    18B Dec 10 20:25 .gitignore
48 -rw-r--r--   1 schulmanr  staff    21K Dec 28 22:02 Cargo.lock
 8 -rw-r--r--   1 schulmanr  staff   197B Dec 28 22:01 Cargo.toml
 0 drwxr-xr-x   4 schulmanr  staff   136B Dec 10 21:39 resources
 0 drwxr-xr-x   6 schulmanr  staff   204B Dec 28 22:14 src
 0 drwxr-xr-x   3 schulmanr  staff   102B Nov 22 10:30 target
```

**Comment by rschulman** [comment]
Through some println debugging, the error is happening when configuring the watchers, around line 86 of main.rs.

**Comment by rschulman** [comment]
Its because config::WATCH_DIRS lists a bunch of directories that don't necessarily exist and FsEvent.watch is choking on it not existing...

**Comment by passcod** [comment]
Hmm, that's a Notify bug.

**Comment by mfolkeseth** [comment]
Having the same problem

**Comment by Np2x** [comment]
Closed by github.com/

**Comment by dwijnand** [comment]
Cargo installing this currently still presents the same backtrace, because the "latest commit" [only updated to notify 3.0.1](https://github.com/passcod/cargo-watch/commit/2a9c9a9dade2c4aa4b65c6084d87dd4000b59775#diff-80398c5faae3c069e4e6aa2ed11b28c0R35), while the [fix](https://github.com/passcod/notify/commit/1674bfad8e76fd3351a2156ab225d0b698f112aa#diff-4ac32a78649ca5bdd8e0ba38b7006a1eL6) seems to be scheduled for 4.0.0, which is [currently unreleased](https://github.com/rust-lang/crates.io-index/blob/4881d57cd282d28c0327fc53cc8aab86ba2ad4ab/no/ti/notify) (if I understand cargo/crates.io properly).

Unfortunately, using the latest "notify" locally to build cargo-watch:

```diff
diff --git a/Cargo.toml b/Cargo.toml
index fb06d7d..37be97c 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -32,7 +32,7 @@ env_logger = "0.3.5"
 lazy_static = "0.2.2"
 libc = "0.2.18"
 log = "0.3.6"
-notify = "3.0.1"
+notify = { path = "../notify", verrsion = "4.0.0" }
 regex = "0.1.80"
 rustc-serialize = "0.3.22"
 wait-timeout = "0.1.3"
```

then launching `cargo watch build` no longer panics, but it doesn't work :-( :

```
23:30:04 $ cargo clean

23:30:07 $ cargo build
   Compiling libc v0.2.17
   Compiling bitflags v0.7.0
   Compiling unicode-normalization v0.1.2
   Compiling gcc v0.3.38
   Compiling matches v0.1.4
   Compiling pkg-config v0.3.8
   Compiling unicode-bidi v0.2.3
   Compiling openssl-sys v0.9.1
   Compiling idna v0.1.0
   Compiling cmake v0.1.18
   Compiling libz-sys v1.0.7
   Compiling curl-sys v0.3.4
   Compiling libssh2-sys v0.2.4
   Compiling libgit2-sys v0.6.4
   Compiling url v1.2.3
   Compiling git2 v0.6.2
   Compiling ggh v0.1.0 (file:///d/ggh)
    Finished debug [unoptimized + debuginfo] target(s) in 17.0 secs

23:30:26 $ cargo clean

23:30:28 $ cargo watch build
Waiting for changes... Hit Ctrl-C to stop.
^C

23:30:45 !
```

**Comment by Np2x** [comment]
Uff you're right sorry for being not careful enough. I created a new pull request (https://github.com/passcod/notify/pull/111) which should fix the problem properly.

**Comment by passcod** [comment]
Closed with v3.1.2

**Comment by dwijnand** [comment]
Works great, thank you.
